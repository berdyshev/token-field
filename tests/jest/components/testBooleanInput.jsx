import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import BooleanInput from 'Components/Input/BooleanInput';
import { noop } from '@deskpro/react-components/dist/utils';

Enzyme.configure({ adapter: new Adapter() });

it('+++capturing Snapshot of BooleanInput with true set', () => {
  const token = {
    type:  'user-message',
    value: true
  };
  const renderedValue = renderer.create(
    <BooleanInput
      token={token}
      className="test"
      selectPreviousToken={noop}
      selectNextToken={noop}
      removeToken={noop}
    />
  ).toJSON();
  expect(renderedValue).toMatchSnapshot();
});
it('+++capturing Snapshot of BooleanInput with false set', () => {
  const token = {
    type:  'user-message',
    value: false
  };
  const renderedValue = renderer.create(
    <BooleanInput
      token={token}
      className="test"
      selectPreviousToken={noop}
      selectNextToken={noop}
      removeToken={noop}
    />
  ).toJSON();
  expect(renderedValue).toMatchSnapshot();
});
it('+++capturing Snapshot of BooleanInput with empty value', () => {
  const token = {
    type:  'user-message',
    value: null
  };
  const renderedValue = renderer.create(
    <BooleanInput
      token={token}
      className="test"
      selectPreviousToken={noop}
      selectNextToken={noop}
      removeToken={noop}
    />
  ).toJSON();
  expect(renderedValue).toMatchSnapshot();
});
