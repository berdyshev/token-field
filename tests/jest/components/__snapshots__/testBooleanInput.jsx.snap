// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`+++capturing Snapshot of BooleanInput with empty value 1`] = `
<div
  className="token test"
>
  <div>
    <div
      className="dp-code label"
    >
      :
    </div>
    <span
      className="value"
      onClick={[Function]}
    >
      ________
    </span>
  </div>
  <div
    className="token-remove remove"
    onClick={[Function]}
  >
    X
  </div>
</div>
`;

exports[`+++capturing Snapshot of BooleanInput with false set 1`] = `
<div
  className="token test"
>
  <div>
    <div
      className="dp-code label"
    >
      :
    </div>
    <span
      className="value"
      onClick={[Function]}
    >
      No
    </span>
  </div>
  <div
    className="token-remove remove"
    onClick={[Function]}
  >
    X
  </div>
</div>
`;

exports[`+++capturing Snapshot of BooleanInput with true set 1`] = `
<div
  className="token test"
>
  <div>
    <div
      className="dp-code label"
    >
      :
    </div>
    <span
      className="value"
      onClick={[Function]}
    >
      Yes
    </span>
  </div>
  <div
    className="token-remove remove"
    onClick={[Function]}
  >
    X
  </div>
</div>
`;
